Client/game_sa/CWeaponSA.cpp

```diff
@@ -20,230 +19,83 @@
 
 extern CGameSA* pGame;
 
-CWeaponSA::CWeaponSA(CWeaponSAInterface* pInterface, CPed* pOwner, eWeaponSlot weaponSlot)
-    : m_pInterface{pInterface}, m_pOwner{pOwner}, m_weaponSlot{weaponSlot}
+CWeaponInfo* CWeaponSA::GetInfo(const eWeaponSkill& skill) const
 {
-}
-
-void CWeaponSA::Destroy()
-{
-    if (!m_pOwner)
-    {
-        delete m_pInterface;
-        delete this;
-    }
-}
-
-eWeaponType CWeaponSA::GetType()
-{
-    return m_pInterface->m_eWeaponType;
-};
-
-void CWeaponSA::SetType(eWeaponType type)
-{
-    m_pInterface->m_eWeaponType = type;
-}
-
-eWeaponState CWeaponSA::GetState()
-{
-    return m_pInterface->m_eState;
-}
-
-void CWeaponSA::SetState(eWeaponState state)
-{
-    m_pInterface->m_eState = state;
-}
-
-DWORD CWeaponSA::GetAmmoInClip()
-{
-    return m_pInterface->m_nAmmoInClip;
-}
-
-void CWeaponSA::SetAmmoInClip(DWORD dwAmmoInClip)
-{
-    m_pInterface->m_nAmmoInClip = dwAmmoInClip;
-}
-
-DWORD CWeaponSA::GetAmmoTotal()
-{
-    return m_pInterface->m_nAmmoTotal;
-}
-
-void CWeaponSA::SetAmmoTotal(DWORD dwAmmoTotal)
-{
-    m_pInterface->m_nAmmoTotal = dwAmmoTotal;
-}
-
-CPed* CWeaponSA::GetPed()
-{
-    return m_pOwner;
-}
-
-eWeaponSlot CWeaponSA::GetSlot()
-{
-    return m_weaponSlot;
+    return m_interface ? pGame->GetWeaponInfo(m_interface->m_eWeaponType, skill) : nullptr;
```

Comment:
Is it possible that m_interface is null?

Reply:
I've discussed the situation around "Is it possible that (x) is null?" cases with him on Discord, there are particular security reasons for additional checks that may seem redundant.